staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go

```diff
@@ -505,6 +513,8 @@ type DataSourceSource struct {
 	PVC *DataVolumeSourcePVC `json:"pvc,omitempty"`
 	// +optional
 	Snapshot *DataVolumeSourceSnapshot `json:"snapshot,omitempty"`
+	// +optional
+	DataSource *DataSourceRefSourceDataSource `json:"dataSource,omitempty"` 
```

Comment:
Could be nice to add a few words about max depth/self reference here so they show up in `kubectl explain`